library/Fission/Web/DNS.hs

```diff
@@ -3,35 +3,36 @@ module Fission.Web.DNS
   , server
   ) where
 
+import           Database.Esqueleto
+import           Servant
 
 import qualified Network.AWS.Auth    as AWS
-import           Servant
+import qualified Network.IPFS.Types as IPFS
+import           Network.IPFS.CID.Types
 
+import           Fission.Models
 import           Fission.Prelude
 
 import           Fission.AWS.Route53
 import qualified Fission.AWS.Types   as AWS
 
-import           Fission.User      as User
-import           Fission.Web.Server
 import           Fission.Web.Error as Web.Err
 
-import qualified Network.IPFS.Types as IPFS
-import           Network.IPFS.CID.Types
-
 type API = Capture "cid" CID
         :> PutAccepted '[PlainText, OctetStream] AWS.DomainName
 
 server ::
-  ( HasLogFunc         cfg
-  , Has IPFS.Gateway   cfg
+  ( Has IPFS.Gateway   cfg
   , Has AWS.AccessKey  cfg
   , Has AWS.SecretKey  cfg
   , Has AWS.ZoneID     cfg
   , Has AWS.DomainName cfg
+  , MonadReader        cfg m
+  , MonadLogger            m
+  , MonadUnliftIO          m
+  , MonadThrow             m
   )
-  => User
-  -> RIOServer cfg API
-server User { username } cid = 
-  registerDomain username cid
-    >>= Web.Err.ensureM
+  => Entity User
+  -> ServerT API m
```

Comment:
Now that we're using the `Monad*` capabilities, we're not specialized to RIO. There's no reason for `RIOServer` to exist anymore, so we're back to the native `ServerT api m`.

For context: 

```haskell
newtype RIOServer cfg api = ServerT api (RIO cfg)
```